src/vs/workbench/contrib/terminal/browser/terminalEditorService.ts

```diff
@@ -153,31 +165,71 @@ export class TerminalEditorService extends Disposable implements ITerminalEditor
 		input.setGroup(editorPane?.group);
 	}
 
-	getOrCreateEditorInput(instance: ITerminalInstance | SerializedTerminalEditorInput, isFutureSplit: boolean = false): TerminalEditorInput {
-		let cachedEditor;
-		if ('id' in instance) {
-			cachedEditor = this._editorInputs.get(instance.id);
-		} else if ('instanceId' in instance) {
-			cachedEditor = this._editorInputs.get(instance.instanceId);
+	getOrCreateEditorInput(instanceOrUri: ITerminalInstance | SerializedTerminalEditorInput | URI, isFutureSplit: boolean = false): TerminalEditorInput {
+		let resource: URI;
+		if (URI.isUri(instanceOrUri)) {
+			resource = instanceOrUri;
+		} else if ('instanceId' in instanceOrUri) {
+			resource = instanceOrUri.resource;
+		} else {
+			resource = URI.parse(instanceOrUri.resource);
 		}
+		const inputKey = resource.path;
+		const cachedEditor = this._editorInputs.get(inputKey);
 		if (cachedEditor) {
 			return cachedEditor;
 		}
 
-		if ('pid' in instance) {
-			instance = this._terminalInstanceService.createInstance({ attachPersistentProcess: instance }, TerminalLocation.Editor);
+		if ('pid' in instanceOrUri) {
+			instanceOrUri = this._terminalInstanceService.createInstance({ attachPersistentProcess: instanceOrUri }, TerminalLocation.Editor);
+		}
+
+		// Terminal from a different window
+		if (URI.isUri(instanceOrUri)) {
+			const terminalIdentifier = parseTerminalUri(instanceOrUri);
+			if (terminalIdentifier.instanceId) {
+				this._primaryOffProcessTerminalService.requestDetachInstance(terminalIdentifier.workspaceId, terminalIdentifier.instanceId).then(attachPersistentProcess => {
+					const instance = this._terminalInstanceService.createInstance({ attachPersistentProcess }, TerminalLocation.Editor, resource);
+					input.setTerminalInstance(instance);
+					// TODO: fix and use an event instead, for now is used to trigger TerminalEditor's setInput,
+					// which attaches to the element
```

Comment:
Let's replace this with an explanatory comment and remove the on did attach event if we don't need it anymore